tests/runtime/test_http_server.py

```diff
@@ -267,6 +271,42 @@ def test_model_not_found():
         assert response.status_code == 404
 
 
+def test_pydantic_wrapping_with_enums():
+    """Check that the pydantic wrapping works on our data models when they have enums"""
+    # The NLP GeneratedTextStreamResult data model contains enums
+
+    # Check that our data model is fine and dandy
+    token = GeneratedToken(text="foo")
+    assert token.text == "foo"
+
+    # Wrap the containing data model in pydantic
+    http_server.RuntimeHTTPServer._dataobject_to_pydantic(GeneratedTextStreamResult)
+
+    # Check that our data model is _still_ fine and dandy
+    token = GeneratedToken(text="foo")
+    assert token.text == "foo"
+
+
+def test_pydantic_wrapping_with_lists():
+    """Check that pydantic wrapping works on data models with lists"""
+
+    @dataobject(package="test.test")
+    class Bar(DataObjectBase):
+        baz: int
+
+    @dataobject(package="test.test")
+    class Foo(DataObjectBase):
```

Comment:
Uhh the error (from what I understand) is that we're trying to use `class Foo` within the same descriptor pool as other tests, specifically `test_data_stream_source` (we create a different descriptor pool for each of the the `dataobject` tests).

Reply:
I guess a simple way would be to just rename this class, the ideal way would be to make the `temp_dpool` as a fixture that can be reused by multiple test files?

Reply:
Not worth fixing for this test imo, although I _would_ like to do that as an extra action item so that we can freely declare same-name `dataobjects` in the future in other test classes?

Reply:
@prashantgupta24 yeah, it would also be nice if py-to-proto allowed us to do this by package instead of requiring a temporary descriptor pool. I think this is potentially a problem if two dataobjects in different packages have the same name. I don't know if I'm convinced that we should say that datamodel classes need to have globally unique names.

Reply:
Yeah that's a good point as well!